syntax = "proto3";

package feedranker;

message RankRequest {
  string user_id = 1;
  repeated string item_ids = 2;
  map<string, float> user_features = 3;
  map<string, float> context_features = 4;
}

message RankedItem {
  string item_id = 1;
  float score = 2;
}

message RankResponse {
  repeated RankedItem ranked_items = 1;
}

service FeedRanker {
  rpc Rank(RankRequest) returns (RankResponse);
}
